osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Bindables;
+using System.Collections.Generic;
+using osu.Game.Rulesets;
+using osuTK;
+using osu.Game.Rulesets.UI;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osuTK.Graphics;
+using System;
+using System.Linq;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class LeaderboardModSelector : Container
+    {
+        public Bindable<IEnumerable<Mod>> SelectedMods = new Bindable<IEnumerable<Mod>>();
+
+        private RulesetInfo ruleset = new RulesetInfo();
+        private readonly FillFlowContainer<SelectableModIcon> modsContainer;
+
+        public LeaderboardModSelector()
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            Child = modsContainer = new FillFlowContainer<SelectableModIcon>
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        public void ResetRuleset(RulesetInfo ruleset)
+        {
+            SelectedMods.Value = new List<Mod>();
+
+            if (ruleset is null)
+            {
+                this.ruleset = null;
+                modsContainer.Clear();
+                return;
+            }
+
+            if (this.ruleset?.Equals(ruleset) ?? false)
+            {
+                deselectAll();
+                return;
+            }
+
+            this.ruleset = ruleset;
+
+            modsContainer.Clear();
+
+            foreach (var mod in ruleset.CreateInstance().GetAllMods())
+            {
+                if (mod.Ranked)
+                    modsContainer.Add(new SelectableModIcon(mod));
+            }
+
+            foreach (var mod in modsContainer)
+                mod.OnSelectionChanged += selectionChanged;
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            var mods = SelectedMods.Value?.ToList() ?? new List<Mod>();
```

Comment:
why are you null checking in here, isn't selected mods already have a list when ruleset is reset?